data/campaigns/Sceptre_of_Fire/scenarios/9_Caverns_of_Flame.cfg

```diff
@@ -664,7 +669,6 @@
         {CHANGE_MAP 4}
         {CHANGE_MAP 5}
         {CHANGE_MAP 6}
-        {CHANGE_MAP 7}
```

Comment:
In the last scenario you have to make the volcano erupt by activating a rune. Each one of those {CHANGE_MAP} makes the lava flow and kill all units it encounters.
On the 7th turn the volcano is supposed to erupt, killing everyone and making you win.
Because of {CHANGE_MAP 7}, the flowing lava may kill some hero, triggering a defeat event; but, at the same time, the volcano will erupt etc... and since you have lost a hero, you won't win but you'll lose, even if you should not.

Reply:
o.O Good catch.
If you want to still use the 7th overlay for visual reasons … one could add a check in the macros for the death events, to only cause defeat if _overlayturn_ is not 7. That check would be in every scenario, but that's fine.
Another option to still use the overlay would be to postpone the eruption to overlayturn 8
If the 7th map file is not used anymore instead like now, then it could be deleted.

PS: I think with an switch-case for CHANGE_MAP the code would be more readable.

Reply:
> If you want to still use the 7th overlay for visual reasons … one could add a check in the macros for the death events, to only cause defeat if overlayturn is not 7. That check would be in every scenario, but that's fine

I've tried doing what you are suggesting here. Imho, there is a very little difference in terms of visual effects, at the cost of more and more lines of code. 

> If the 7th map file is not used anymore instead like now, then it could be deleted.

As you say, i was even wondering if that overlay's file could be removed... but let me know what you think.

> Another option to still use the overlay would be to postpone the eruption to overlayturn 8

This would be a good solution. Anyway, imo the last scenario needs some rework and maybe rebalancement, and some new prose too (objectives): i can't do all of these things now, since they would require a lot of time to me.

Reply:
> PS: I think with an switch-case for CHANGE_MAP the code would be more readable.

I thougth a lot about some code refactoring for SoF, since there are lots of repetitions, few macros (that could help shortening the files) and things like the one you've just pointed out.
But i think that campaigns are the best gym for the ones who want to learn WML, and having different syntactic examples of how one thing can be achieved is fantastic (in fact, i started by copy-pasting lines from TRoW and SotBE... sigh).
Anyway, this is just an opinion of mine, and i agree with you that a [switch] tag would do better there

Reply:
> I've tried doing what you are suggesting here. Imho, there is a very little difference in terms of visual effects, at the cost of more and more lines of code.

Sounds reasonable

> As you say, i was even wondering if that overlay's file could be removed... but let me know what you think.
…
This would be a good solution. Anyway, imo the last scenario needs some rework and maybe rebalancement, and some new prose too (objectives): i can't do all of these things now, since they would require a lot of time to me.

I would keep things for now like they are – so this can be merged. I assume would it is from the balance fine if the scenario is 1 turn longer and tend to let it end in overlayturn 8. However, it's rather a _visual question_.
One more technical option came to mind, killing the unit with `fire_event=no` in overlayturn 7.

That turn doesn't really matter (I assume) and having one more or less lava increase is not important either… -> whatever works out good / looks good.

Things take time and there is actually no pressure to do things.
For refactorings, they have usually not a direct payoff, I sometimes change them as I encounter them while doing other things (and sometimes not)
Having different ways how to do some things is nice (i.e. TSG includes the death events in another way).